Added script to tally and analyze GitHub issue comments from JSON data

JasonUranta opened the pull request:
Fixes #23
Fixes #4 

This PR introduces tally-issue-comments.py, a Python script that processes the JSON output from the GitHub REST API to analyze issue comment data. The script counts the total number of unique issues, calculates the total number of comments across all issues, identifies the issue with the most comments, and presents this information in a formatted table showing comment counts per issue. The output provides clear summary statistics followed by a detailed breakdown of comments for each issue number.

Commit:
committing to make sure files are updated in GitGitHub as I start issue #23 now

Commit:
Finished Issue #23: tallied the total # of issues and comments, made a table to reprsent the number of comments per issue, and identified the issue with the most comments.

Commit:
checking

Commit:
comming fr fr

chinaexpert1 reviewed (dismissed):
I am not sure what to do about this PR. It says 1554 files?? There are a lot of libraries being uploaded here and it would be nice if Jason did a separate PR for the libararies and his work, so I don't have to pick through this. As it is it is too much to go through.

On another note, if Jason is now comfortable doing his code reviews with both ChatGPT and the [Code Review check list ](https://github.com/hackforla/data-science/wiki/Self%E2%80%90Review-Checklist-for-Code-Commits) then there isn't much I can add as a reviewer anymore because he is a more experienced coder than I am. I usually just use ChatGPT. I think it is a better idea if Sandy is his reviewer.

At any rate, I am just approving this PR because it is too big to review. We can merge it if Bonnie is ok with that.

Commit:
added feature to produce markdown file version of data along with JSON version

Commit:
Add .gitignore and remove binary/venv files

JasonUranta commented:
<details><summary>Instructions</summary>

1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (if necessary): "Add any pictures that will help illustrate what you are working on."
</details> 

1. Progress: Fixed the error where I added my venv metadata in the PR to be pushed. Added a feature to create an MD file as part of the output, along with the JSON file, of the fetch-issue-comments.py script to allow for better tokenization.
2. Blockers: N/A
3. Availability: N/A
4. ETA: N/A
5. Pictures (if necessary): N/A

sandy3w commented:
Hey Jason,

The script looks really clean and good to go, just some minor confusions on my side.

What is the difference between issue-comments.json and issue_comments.json? I see the YAML file writing to issue-comments.json but the tally script reading from issue_comments.json.

Also, is the script meant to stay as a command line script or does the produced data have chances to be reused?

In addition, I see a lot of parameters in the comments data files, are all of them required or can the data be cleaned up a bit (e.g. removing reactions).

Other than these everything looks good!

Best,
Sandy

sandy3w reviewed (changes requested):
Hey Jason,

The script looks really clean and good to go, just some minor confusions on my side.

What is the difference between issue-comments.json and issue_comments.json? I see the YAML file writing to issue-comments.json but the tally script reading from issue_comments.json.

Also, is the script meant to stay as a command line script or does the produced data have chances to be reused?

In addition, I see a lot of parameters in the comments data files, are all of them required or can the data be cleaned up a bit (e.g. removing reactions).

Other than these everything looks good!

Best,
Sandy

Commit:
updated and relabeled YAML and python to include input meta data in JSON and md as well as timestamps in file names and cleaned up directory (re-commiting without personal access token)

sandy3w reviewed (commented):
Heyyy Jason, 

I think the script is a bit outside of my coding knowledge, but everything looks great from what I can understand lol. The cleaned up version of the comments md file looks nice and seems to definitely save lots of space!

Perhaps just rename some of the files to be a bit more clear?

Otherwise, nothing from me:)

chinaexpert1 commented:
Hi Jason,
Remember Sally is your reviewer now. Can you cancel your request please?
Andrew

On Mon, Oct 27, 2025, 7:07 PM Jason Uranta ***@***.***> wrote:

> @JasonUranta <https://github.com/JasonUranta> requested your review on:
> #34 <https://github.com/hackforla/ai-skills-assessor/pull/34> Added
> script to tally and analyze GitHub issue comments from JSON data.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackforla/ai-skills-assessor/pull/34#event-20540657287>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA6SRH32PWH7J3DKHZUXXOD3Z3FWXAVCNFSM6AAAAACJXOYUD2VHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMRQGU2DANRVG4ZDQNY>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

JasonUranta commented:
@chinaexpert1 
My bad, removed your review request. 


@sandy3w 
The output files are named using time stamps at Bonnie's request, and I think the python and YAML files are named according to the actions they perform. If you have any suggestions on better naming practices, please let me know! Otherwise, I think this PR is officially ready to be merged!

If you agree with me about about merging this PR, can you resolve the "review changes" request and instead approve the changes so I can merge the PR?

JasonUranta commented:
<details><summary>Instructions</summary>

1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (if necessary): "Add any pictures that will help illustrate what you are working on."
</details> 

1. Progress: Finally finished all review requests on PR #34 for this issue, and it should be merged soon.
2. Blockers:  Waiting on @sandy3w to resolve the "changes requested" marker on PR #34 so I can merge it
3. Availability: Low availability this week due to Thanksgiving, but free next.
4. ETA: N/A
5. Pictures (if necessary): N/A
